src/main/java/net/sf/rails/game/GameManager.java

```diff
@@ -1200,10 +1204,8 @@ public boolean getGameOverReportedUI(){
         List<String> b = new ArrayList<String>();
 
         /* Sort players by total worth */
-        List<Player> rankedPlayers = new ArrayList<Player>();
-        for (Player player : getRoot().getPlayerManager().getPlayers()) {
-            rankedPlayers.add(player);
-        }
+        List<Player> rankedPlayers = new ArrayList<>();
+        rankedPlayers.addAll(getRoot().getPlayerManager().getPlayers());
```

Comment:
Use of new array operations such as `addAll()` vs iteration

Reply:
Maybe even better:

```java
List<Player> rankedPlayers = getRoot().getPlayerManager().getPlayers().stream().sorted().collect(Collectors.toList());
```

Reply:
This chained call is really hard to read and probably to understand..
Of course i am no formally educated programmer...

Reply:
I also find chained calls hard to read and I am a formally trained developer! That is why I have formatting rules set up to automatically reformat something like the above into something like this:
```
List<Player> rankedPlayers = getRoot().
    getPlayerManager().
    getPlayers().
    stream().
    sorted().
    collect(Collectors.toList());
```
Streams (or chains actually) are initially a bit more difficult to understand but once you learn to mental "parse" them they actually can be much easier to understand than the equiv written out verbosely.

In this case though the use of `addAll()` which is cleaner is likely more appropriate than a `stream()` as the original code had no sorting, ie we are just looking at these examples as direct equivalences simply for example purposes.

Reply:
> In this case though the use of addAll() which is cleaner is likely more appropriate than a stream() as the original code had no sorting, ie we are just looking at these examples as direct equivalences simply for example purposes.

But the code has sorting, see:

https://github.com/jklap/Rails/blob/4937f172666931c233b6518cee8183cf8608e18a/src/main/java/net/sf/rails/game/GameManager.java#L1207-L1209

Reply:
This is why I hate GitHubs code review tool-- too easy to miss the code below the comments!

Yes, either of these solutions work-- and not really sure one is any better than the other. I tend to use the non-stream version here and use `addAll()` and then `Collections.sort()` because I find it easier for others to understand but that is just a personal preference and that in this case both are completely viable solutions